lib/client.js

```diff
@@ -263,7 +263,23 @@ Client.prototype._buildForward = function() {
       }
 
       if (qos && options.messageId) {
-        that.server.updateOfflinePacket(that, options.messageId, packet, doForward);
+
+        //converting the payload to a JSON object
+        var packetPayload = JSON.parse(packet.payload.toString());
```

Comment:
How do you know that payload is always a valid JSON? 
it has also introducing considerable processing overhead in heavy deployments